content/en/docs/how-tos/adding-a-cluster-profile.md

```diff
@@ -90,3 +90,77 @@ changes.  These are:
 - A `ConfigMap` is required for the profile.  The convention is for it to be
   named `cluster-profile-<name>`, in which case a single new `switch` label is
   required.
+
+## VPN connection
+
+For platforms that need access to restricted environments, `ci-operator`
+supports adding a dedicated VPN connection to each test step.  Since this is a
+requirement of specific platforms, it is triggered by special files in the
+[cluster profile(s)]({{< ref "docs/how-tos/adding-a-cluster-profile#what-is-a-cluster-profile" >}})
+associated with those platforms.  This process is transparent to the test
+command: when a VPN connection is requested at the test level, it is set up
+automatically by the test platform with no changes required to individual tests.
+
+{{< alert title="Note" color="info" >}}
+Details of the implementation can be found in the
+<a href="https://docs.google.com/document/d/1mPjrHVS1EvmLdq4kGhRazTpGu6xVZDyGpVAphVZhX4w/edit?resourcekey=0-KA-qXXq1J2bTR7o6Kit9Vw">design document</a>.
+{{< /alert >}}
+
+### Cluster profile
+
+VPN connections are requested by the presence of a special file named `vpn.yaml`
+in the cluster profile, detected when the test steps are about to be executed.
+This file should have the following fields:
+
+- `image`: _pull spec_ of the image to be used for the VPN client container.
+  This image should contain the packages necessary to establish the connection,
+  as well as the `bash` shell to execute the container's entry point script.
+- `commands`: the name of another file in the cluster profile (e.g.
+  `openvpn.sh`) which contains the VPN client's entry point script.  This script
+  is effectively executed as `bash -c "$(<f")"`, where `f` is the value
+  associated with the `commands` key.
+- `wait_timeout`: the maximum amount of time the step script should wait before
```

Comment:
Might be another future work: tests can opt in the following action upon timeout: fail the test or run the test without VPN.

Reply:
This is a bit of hack, I'll admit.

The problem here is this wraps `entrypoint`, which is what signals that the test has terminated.  So failing here would interfere with the other Prow components.  The test presumably will fail immediately in the absence of a VPN connection (and a warning is emitted right at the top of the test logs), so I don't expect it to be a problem, I plan to improve this in the future.